### ISSUE TITLE
shouldBe throws KotlinReflectionInternalError on java enums with property override

### ISSUE THREAD

**Issue by scharrenberg** [kotest/kotest#1611]
Hi,

thanks for this pleasant to work with library that is under constant improvement. When upgrading Kotest in one of our projects I stumbled upon the following behaviour.

When comparing two objects and one of those objects is a Java enum value which overrides a property then `shouldBe` throws `kotlin.reflect.jvm.internal.KotlinReflectionInternalError: Unresolved class: class [...]`

**Which version of Kotest are you using**
The replication works as of 4.1.0

Minimal replication:

```kotlin
import io.kotest.matchers.shouldBe
import org.junit.jupiter.api.Test

class Replication {

    @Test
    fun `Pass with default property`() {
        ThingWithProperty.SimpleThingWithProperty.shouldBe(ThingWithProperty.SimpleThingWithProperty)
    }

    @Test
    fun `Fail because of property override`() {
        ThingWithProperty.ThingWithOverriddenProperty.shouldBe(ThingWithProperty.ThingWithOverriddenProperty)
    }
}
```

ThingWithProperty.java
```java
public enum ThingWithProperty {

    SimpleThingWithProperty,

    ThingWithOverriddenProperty {
        @Override
        public String getGreeting() {
            return "Hi";
        }
    };

    public String getGreeting() {
        return "Hello";
    }
}
```

**Comment by LeoColman** [comment]
Could you please confirm that the Java file is actually in the classpath when you compile? It seems to be failing to pickup a class

**Comment by sksamuel** [comment]
I can reproduce.

com/sksamuel/kotest/specs/funspec/ThingWithProperty
java.lang.NoClassDefFoundError: com/sksamuel/kotest/specs/funspec/ThingWithProperty

It seems to me that .java files are not picked up when in kotlin source roots. That wouldn't be a kotest thing, but would just need to configure gradle.

**Comment by scharrenberg** [comment]
@LeoColman This does not seem like a build configuration issue to me. Please see https://github.com/scharrenberg/ktest-bug

When building with Gradle `Pass with default property` passes although it references the Java enum whereas `Fail because of property override` fails.

**Comment by sksamuel** [comment]
I've found the bug, it seems to be a kotlin issue actually. It throws if you call `.isData` on a KClass that is a java enum. Easy enough to work around though, I will add a fix for 4.2.1

**Comment by sksamuel** [comment]
Fix in master.
